lang/java/avro/src/main/java/org/apache/avro/data/Json.java

```diff
@@ -55,8 +56,12 @@ private Json() {}                               // singleton: no public ctor
   public static final Schema SCHEMA;
   static {
     try {
```

Comment:
Do we have to support Java 1.6 in Avro, or we require 1.7+? I'm wondering if we can use try-with-resource statements instead of closing the stream in finally. Also, can we merge the two nested try{} blocks into one?